### PR TITLE
CAMEL-18617: Some health checks are hidden when running withg supervised controller enabled

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

File: components/camel-microprofile/camel-microprofile-health/src/test/java/org/apache/camel/microprofile/health/CamelMicroProfileHealthSupervisedRoutesMainTest.java
Comment (lburgazzoli) [inline components/camel-microprofile/camel-microprofile-health/src/test/java/org/apache/camel/microprofile/health/CamelMicroProfileHealthSupervisedRoutesMainTest.java:59]: @jamesnetherton I amended only this test to include also components checks vs supervising controller, not sure if that's enough for the other cases

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java
Comment (davsclaus) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java:33]: Avoid importing impl classes as they should not be used in the public

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java
Comment (davsclaus) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java:56]: Can you use HealthCheckRepository API instead to avoid impl import

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java
Comment (davsclaus) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java:127]: Lookup the repository without impl import

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java
Comment (davsclaus) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java:41]: Same here as in the consumer

**Review by davsclaus — CHANGES_REQUESTED** [review]

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java
Comment (lburgazzoli) [reply components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java:33]: this is because the ComponentsHealthCheckRepository has an add/remove health check methods: should we create a WritableHealthCheckRepository interface ?

File: components/camel-microprofile/camel-microprofile-health/src/test/java/org/apache/camel/microprofile/health/CamelMicroProfileHealthSupervisedRoutesMainTest.java
Comment (jamesnetherton) [reply components/camel-microprofile/camel-microprofile-health/src/test/java/org/apache/camel/microprofile/health/CamelMicroProfileHealthSupervisedRoutesMainTest.java:59]: It's probably ok. Would be nice to cover the non-supervised scenario if possible. I'll leave it up to you whether you want to do that.

Comment (lburgazzoli) [reply components/camel-microprofile/camel-microprofile-health/src/test/java/org/apache/camel/microprofile/health/CamelMicroProfileHealthSupervisedRoutesMainTest.java:59]: Will do, do you have any preference about where this should be done ?

Comment (jamesnetherton) [reply components/camel-microprofile/camel-microprofile-health/src/test/java/org/apache/camel/microprofile/health/CamelMicroProfileHealthSupervisedRoutesMainTest.java:59]: Might be enough to clone `CamelMicroProfileHealthConsumerTest` and modify it a bit.

**Review by oscerd — APPROVED** [review]

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java
Comment (lburgazzoli) [reply components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java]: Pushed a new version, however the Kafka Consumer/Producer HealthCheck  are also dependant on on an impl `org.apache.camel.impl.health.AbstractHealthCheck`

File: components/camel-microprofile/camel-microprofile-health/src/test/java/org/apache/camel/microprofile/health/CamelMicroProfileHealthSupervisedRoutesMainTest.java
Comment (lburgazzoli) [reply components/camel-microprofile/camel-microprofile-health/src/test/java/org/apache/camel/microprofile/health/CamelMicroProfileHealthSupervisedRoutesMainTest.java:44]: I ended up creating a new test as it impacted the current tests as they do assertion based on the number of checks

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java
Comment (davsclaus) [reply components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java]: yeah the entire classes in camel-health should be moved to another package - as org.apache.camel.impl is for the core routing engine and camel itself.

**Review by jamesnetherton — APPROVED** [review]

**Comment by davsclaus** [comment]
It may be that some health checks in aws has been migrated as well, so updates are likely needed there too. This can be migrated in another PR

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java
Comment (davsclaus) [reply components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java]: The `impl` is also in camel-cluster / camel-cloud so its more general. Therefore in light of this then its okay in 3.x to import the impl - We can revisit this in 4.0

**Comment by lburgazzoli** [comment]
> It may be that some health checks in aws has been migrated as well, so updates are likely needed there too. This can be migrated in another PR

ok, will follow up once this gets merged

**Comment by lburgazzoli** [comment]
Can anyone help me understanding the test report ? 

```
Logging test to /home/runner/work/camel/camel/automated-build-log/core.log
Failed core test test: 1 verified / 1 failed
There is no log file to copy at target/core-test.log
Searching for modified components
It will test the following 4 components:
components/camel-aws
components/camel-kafka
components/camel-microprofile
components/camel-platform-http-vertx
############################################################
Testing component 1 of 4: components/camel-aws
############################################################

Logging test to /home/runner/work/camel/camel/automated-build-log/components-camel-aws.log
Failed components/camel-aws test test: 1 verified / 2 failed
There is no log file to copy at target/camel-aws-test.log
############################################################
Testing component 2 of 4: components/camel-kafka
############################################################

Logging test to /home/runner/work/camel/camel/automated-build-log/components-camel-kafka.log
components/camel-kafka test completed successfully: 2 verified / 2 failed
Copying test log file at target/camel-kafka-test.log to the log directory
############################################################
Testing component 3 of 4: components/camel-microprofile
############################################################

Logging test to /home/runner/work/camel/camel/automated-build-log/components-camel-microprofile.log
components/camel-microprofile test completed successfully: 3 verified / 2 failed
There is no log file to copy at target/camel-microprofile-test.log
############################################################
Testing component 4 of 4: components/camel-platform-http-vertx
############################################################

Logging test to /home/runner/work/camel/camel/automated-build-log/components-camel-platform-http-vertx.log
components/camel-platform-http-vertx test completed successfully: 4 verified / 2 failed
There is no log file to copy at target/camel-platform-http-vertx-test.log
```

I see that as example, there is a failure in the `camel-platform-http-vertx` tests but:
1. the component should not be impacted by the PR
2. the log file I have downloaded reports that all tests are ok

**Comment by lburgazzoli** [comment]
Can anyone help me understanding the test report ? 

```
Logging test to /home/runner/work/camel/camel/automated-build-log/core.log
Failed core test test: 1 verified / 1 failed
There is no log file to copy at target/core-test.log
Searching for modified components
It will test the following 4 components:
components/camel-aws
components/camel-kafka
components/camel-microprofile
components/camel-platform-http-vertx
############################################################
Testing component 1 of 4: components/camel-aws
############################################################

Logging test to /home/runner/work/camel/camel/automated-build-log/components-camel-aws.log
Failed components/camel-aws test test: 1 verified / 2 failed
There is no log file to copy at target/camel-aws-test.log
############################################################
Testing component 2 of 4: components/camel-kafka
############################################################

Logging test to /home/runner/work/camel/camel/automated-build-log/components-camel-kafka.log
components/camel-kafka test completed successfully: 2 verified / 2 failed
Copying test log file at target/camel-kafka-test.log to the log directory
############################################################
Testing component 3 of 4: components/camel-microprofile
############################################################

Logging test to /home/runner/work/camel/camel/automated-build-log/components-camel-microprofile.log
components/camel-microprofile test completed successfully: 3 verified / 2 failed
There is no log file to copy at target/camel-microprofile-test.log
############################################################
Testing component 4 of 4: components/camel-platform-http-vertx
############################################################

Logging test to /home/runner/work/camel/camel/automated-build-log/components-camel-platform-http-vertx.log
components/camel-platform-http-vertx test completed successfully: 4 verified / 2 failed
There is no log file to copy at target/camel-platform-http-vertx-test.log
```

I see that as example, there is a failure in the `camel-platform-http-vertx` tests but:
1. the component should not be impacted by the PR
2. the log file I have downloaded reports that all tests are ok

**Comment by davsclaus** [comment]
LGTM to merge

**Pull request merged**
